internal/pkg/manager.go

```diff
@@ -67,39 +90,97 @@ func (m *ManagerConfig) Run() error {
 			return err
 		}
 
-		m.Logger.Info("received nats message ", "message: ", string(msg.Body))
-
-		if err = m.updateConfigToLatest(); err != nil {
-			m.Logger.Error("failed to update the config", "error", err)
-		}
+		_ = m.processMsg(msg)
 
+		// TODO - @rizzza - √√ on this ack with Tyler. We ack everything? nack is fatal with this driver.
 		msg.Ack()
 	}
 }
 
-func (m *ManagerConfig) updateConfigToLatest() error {
+// processMsg message handler
+func (m ManagerConfig) processMsg(msg *pubsub.Message) error {
+	pubsubMsg := pubsubx.Message{}
+	if err := json.Unmarshal(msg.Body, &pubsubMsg); err != nil {
+		m.Logger.Errorw("failed to process data in msg", zap.Error(err))
+		return err
+	}
+
+	urn, err := urnx.Parse(pubsubMsg.SubjectURN)
+	if err != nil {
+		m.Logger.Errorw("failed to parse pubsub msg urn", zap.String("subjectURN", pubsubMsg.SubjectURN), zap.Error(err))
+		return err
+	}
+
+	lbID := urn.ResourceID.String()
+	if err = m.updateConfigToLatest(viper.GetString("haproxy.config.base"), lbID); err != nil {
+		m.Logger.Errorw("failed to update haproxy config", zap.String("loadbalancer.id", lbID), zap.Error(err))
+		return err
+	}
+
+	return nil
+}
+
+// updateConfigToLatest update the haproxy cfg to either baseline or one requested from lbapi with optional lbID param
+func (m *ManagerConfig) updateConfigToLatest(baseCfgPath string, lbID ...string) error {
+	if len(lbID) > 1 {
+		return fmt.Errorf("optional lbID param must be not set or set to a singular loadbalancer ID")
+	}
+
 	m.Logger.Info("updating the config")
+
 	// load base config
-	cfg, err := parser.New(options.Path(viper.GetString("haproxy.config.base")))
+	cfg, err := parser.New(options.Path(baseCfgPath), options.NoNamedDefaultsFrom)
 	if err != nil {
 		m.Logger.Fatalw("failed to load haproxy base config", "error", err)
 	}
 
-	// get desired state
-	// transform response
-	// merge desired with base
+	if len(lbID) == 1 {
+		// requested a lb id, query lbapi
+		// get desired state
+		lb, err := m.LBClient.GetLoadBalancer(m.Context, lbID[0])
+		if err != nil {
+			return err
+		}
+
+		// query each pool and store the origins
+		for i, port := range lb.Ports {
+			for _, poolID := range port.Pools {
+				p, err := m.LBClient.GetPool(m.Context, poolID)
+				if err != nil {
+					return err
+				}
+
+				data := lbapi.Pool{
+					ID:      poolID,
+					Name:    p.Name,
+					Origins: p.Origins,
+				}
+
+				lb.Ports[i].PoolData = append(port.PoolData, data)
```

Comment:
I think PoolData could be `nil` here?

Reply:
also maybe add a test for that case if so 👍

Reply:
it's a initialized empty slice. Should be good there.